### ISSUE TITLE
PoA -> Substrate: detect and report validator misbehavior when headers are submitted using unsigned transactions

### ISSUE THREAD

**Issue by svyatonik** [paritytech/parity-bridges-common#39]
imagine #34 in unsigned transactions context - we can't penalize validator for spamming tx pool and blocks with good bad' headers AND validator can just do not relay these blocks to other PoA peers => it'll be never reported as malicious using builtin PoA mechanisms. So we need to detect such validators in bridge and report them back to PoA chain.

The easiest fix (suggested by @tomusdrw ): (c) "...should basically just make sure that every header seen on the bridge is gossiped in the network and that should be enough."

**Comment by svyatonik** [comment]
Update: it would be good to use the same mechanism with signed transactions. Most problems with malicious validators will arise when unsigned transactions are used, but if we'll have that mechanism, then we would we simply ignore misbehavior when signed transactions are used?

**Comment by tomusdrw** [comment]
> Update: it would be good to use the same mechanism with signed transaction

Yes, I think in fact it should be simply a separate component, that just listens to on-chain state (no matter how it is changed (signed / unsigned)) and then makes sure the headers are propagated.

**Comment by tomusdrw** [comment]
Note that it should be done not only for headers, but also for finality proofs (justifications).

**Comment by tomusdrw** [comment]
Re-labeling as Substrate<->Substrate bridge feature.

**Comment by serban300** [comment]
Not sure what would be the best solution, but started implementing some primitives. This way it will also be easier to merge them to the `polkadot-staging` branch. And then we'll put all the pieces together.
